src/Stratis.Bitcoin/P2P/PeerAddressManagerBehaviour.cs

```diff
@@ -68,15 +77,32 @@ protected override void AttachCore()
 
         private async Task OnMessageReceivedAsync(INetworkPeer peer, IncomingMessage message)
         {
+            this.logger.LogTrace("({0}:[{1}], {2}:[{3}])", nameof(peer), peer, nameof(message), message);
+
             try
             {
                 if ((this.Mode & PeerAddressManagerBehaviourMode.Advertise) != 0)
                 {
-                    if ((message.Message.Payload is GetAddrPayload) && (!this.sentAddress))
+                    if (message.Message.Payload is GetAddrPayload)
                     {
+                        if (!peer.Inbound)
+                        {
+                            this.logger.LogTrace($"Outbound peer ${peer} sent {nameof(GetAddrPayload)}. Not replying to avoid fingerprinting attack.");
```

Comment:
No interpolated strings please

Also either use remote socket endpoint on the peer is above, or don't mention the peer at all because we already have identification of the peer in the log prefix and the entry log

Same below

Reply:
What? No interpolated strings?

Where is this in policy?

Reply:
https://github.com/stratisproject/StratisBitcoinFullNode/blob/master/Documentation/using-logging.md#interpolated-strings

Reply:
Fair, fixed